### PR TITLE
wip - doswap() & testing/ fixed waittime logic and calcnetwork fee

### PR DESCRIPTION
**Pull request by Thorian1te**

### CODE REVIEW

**Review by mikewyszinski — CHANGES_REQUESTED** [review]
everything looks good, just needs version bump and change log

File: yarn.lock [hunk 1]
```diff
@@ -2692,6 +2692,11 @@
     "@typescript-eslint/types" "4.18.0"
     eslint-visitor-keys "^2.0.0"
 
+"@xchainjs/xchain-client@^0.11.1":
+  version "0.11.3"
```
Comment (mikewyszinski) [inline yarn.lock:2696]: did you mean to update the yarn.lock?

File: packages/xchain-thorchain-amm/__e2e__/thornode-checkTx.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Network } from '@xchainjs/xchain-client'
```
Comment (mikewyszinski) [inline packages/xchain-thorchain-amm/__e2e__/thornode-checkTx.ts:1]: this file name needs to end in *e2e.ts in order to be picked up by 'yarn e2e'

File: packages/xchain-thorchain-amm/src/utils/thornode.ts [hunk 3]
```diff
@@ -172,22 +172,22 @@ export class Thornode {
   public async checkTx(inboundTxHash: string, sourceChain?: Chain): Promise<TxStatus> {
     let txStatus: TxStatus = { stage: TxStage.INBOUND_CHAIN_UNCONFIRMED, seconds: 0 }
     const txData = await this.getTxData(inboundTxHash)
+    console.log(txData.observed_tx?.tx.chain)
```
Comment (mikewyszinski) [inline packages/xchain-thorchain-amm/src/utils/thornode.ts:175]: remove

File: packages/xchain-thorchain-amm/src/utils/thornode.ts [hunk 4]
```diff
@@ -172,22 +172,22 @@ export class Thornode {
...
     const scheduledQueue = await this.getscheduledQueue()
     const scheduledQueueItem = scheduledQueue?.find((item: TxOutItem) => item.in_hash === inboundTxHash)
     const lastBlock = await this.getLastBlock()
     const lastBlockHeight = lastBlock.find((obj) => {
       return obj
     })
-    console.log(txData.observed_tx?.tx)
-    if (txData.observed_tx?.tx == undefined) {
+    if (txData.observed_tx == undefined) {
       console.log(`Stage 1`)
```
Comment (mikewyszinski) [inline packages/xchain-thorchain-amm/src/utils/thornode.ts:183]: remove all console log statements

File: packages/xchain-thorchain-amm/src/utils/thornode.ts [hunk 5]
```diff
@@ -261,29 +254,41 @@ export class Thornode {
...
+        console.log(synthCheck.length)
+        console.log(pool)
+        txStatus.seconds = this.chainAttributes[THORChain].avgBlockTimeInSecs
+      } else {
+        console.log(synthCheck.length)
+        console.log(pool)
+        const ticker = pool[1].split(`.`, 1).toString()
+        const chain = this.getChain(ticker)
+        txStatus.seconds = this.chainAttributes[chain].avgBlockTimeInSecs
+      }
```
Comment (mikewyszinski) [inline packages/xchain-thorchain-amm/src/utils/thornode.ts:275]: this would be cleaner/safer  if we use assetFromString(),

also remove console.log()

**Review by mikewyszinski — CHANGES_REQUESTED** [review]
good work ser!, just a few nitpicks

File: packages/xchain-thorchain-amm/__e2e__/thornode-checkTx.ts (hunk 2 above)
Comment (Thorian1te) [reply packages/xchain-thorchain-amm/__e2e__/thornode-checkTx.ts:1]: oh yes of course

File: packages/xchain-thorchain-amm/src/utils/thornode.ts (hunk 4 above)
Comment (Thorian1te) [reply packages/xchain-thorchain-amm/src/utils/thornode.ts:183]: ok

**Review by mikewyszinski — APPROVED** [review]

**Pull request merged**